pkgs/by-name/st/styluslabs-write/package.nix

```diff
@@ -0,0 +1,99 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  gitMinimal,
+  pkg-config,
+  SDL2,
+  xorg,
+  dbus,
+  libcap,
+  libGL,
+  roboto,
+  imagemagick,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "styluslabs-write";
+  version = "2024-09-26";
+
+  src = fetchFromGitHub {
+    owner = "styluslabs";
+    repo = "Write";
+    rev = "7110e5db87c154d28efd9507be4ab768af7403e0";
+    hash = "sha256-/IYI/trC14IBYtYn9LkjNrMZoCn6jtQIgq0ZqnsLISs=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./sdl2_link_dynamic.patch ];
+
+  preBuild = ''
+    pushd syncscribble
```

Comment:
Have you tried using sourceRoot instead?

Reply:
`sourceRoot` does not work because the `scribbleres/Makefile` needs access to the sibling directories (aka the git submodules).
These statically linked dependencies are also under different licences (mostly MIT), should I set the licence field to multiple values as [done here](https://github.com/NixOS/nixpkgs/blob/e186dd1a34be7a76c6e2c038ab4b759faec32d5d/pkgs/games/shipwright/default.nix#L192)?